dandi/core/digests/dandietag.py

```diff
@@ -0,0 +1,101 @@
+# from https://github.com/girder/django-s3-file-field/blob/master/s3_file_field/_multipart.py
+from hashlib import md5
+import math
+import os
+from typing import List, Optional, Tuple
+
+
+def mb(bytes_size: int) -> int:
+    return bytes_size * 2 ** 20
+
+
+def gb(bytes_size: int) -> int:
+    return bytes_size * 2 ** 30
+
+
+def tb(bytes_size: int) -> int:
+    return bytes_size * 2 ** 40
+
+
+class DandiETag:
+    REGEX = r"[0-9a-f]{32}-\d{1,4}"
+    MAX_STR_LENGTH = 37
```

Comment:
@yarikoptic What is this based on?  Wouldn't the etag suffixes go up through `-10000`, meaning that the regex should end with `\d{1,5}` and the `MAX_STR_LENGTH` should be 38?

Reply:
yes, I miscalculated I guess (was thinking about  `5,000`?) please fix accordingly